addons/mail/static/src/components/chatter/chatter.xml

```diff
@@ -31,13 +31,22 @@
                             threadLocalId="chatter.thread.localId"
                         />
                     </t>
-                    <t t-if="chatter.thread and chatter.hasActivities and chatter.thread.activities.length > 0">
+                    <t t-if="chatter.isActivityBoxVisible">
                         <ActivityBox
                             class="o_Chatter_activityBox"
                             chatterLocalId="chatter.localId"
                         />
                     </t>
                     <t t-if="chatter.threadView">
+                        <t t-if="chatter.threadView.nonEmptyMessages.length === 0 and (chatter.isActivityBoxVisible or chatter.isAttachmentBoxVisible)">
```

Comment:
According to spec I think the condition should actually be "or" between `nonEmptyMessages` and the other conditions too.

But I find it a bit strange to have the separator sometimes appear/disappear, because it looks too much like a title and not like a separator.
So make it OR instead of AND, and then I think PO should test and give feedback according to this.

Reply:
Also I would also make this condition a computed field in chatter model with a name such as  `isMessageSeparatorVisibile`.

Reply:
> According to spec I think the condition should actually be "or" between `nonEmptyMessages` and the other conditions too.
> 
> But I find it a bit strange to have the separator sometimes appear/disappear, because it looks too much like a title and not like a separator.
> So make it OR instead of AND, and then I think PO should test and give feedback according to this.

If we put "or" between nonEmptyMessages and the other conditions too. then it will be look like below SS
![messages](https://user-images.githubusercontent.com/67690691/106269104-0ffcd400-6252-11eb-818e-c55cfaedba51.png)

Reply:
Thank you for the screenshot. I think this is what was requested, but I'm not sure if it looks good, this is why we should ask for feedback.

Reply:
Maybe make a version where it is always visible, and then ask for test by PO it will be easier to judge when it should be shown or not if it is always visible.

Reply:
Also maybe having the separator is good but actually what we want is to change something else, such as the way we display "today", or the style of the separators.

Reply:
> Maybe make a version where it is always visible, and then ask for test by PO it will be easier to judge when it should be shown or not if it is always visible.

Already discussed with @aju-odoo  and @dit-odoo  they decided to add separator when there is message list is empty and there is a attachment and/or activites are displayed.

Reply:
I will check directly with them then. Because currently the purpose of the task is not reached at all.

> When scrolling down on the message list, the "today" separator becomes hidden
> and it's really hard to tell between messages and activities.

=> the exact same issue is still here if only show the separator if the list is empty.